lib/screens/challenges/challenges.dart

```diff
@@ -1,20 +1,36 @@
 import 'package:flutter/material.dart';
 import 'active.dart';
 import 'completed.dart';
+import '../../models/user.dart';
+import '../../models/user_challenge.dart';
+import '../../data/api.dart';
 
 class ChallengeScreen extends StatefulWidget {
+  final User user;
+
+  ChallengeScreen({Key key , this.user});
+
   @override
   State createState() => ChallengeScreenState();
 }
 
 class ChallengeScreenState extends State<ChallengeScreen> {
-  final List<Widget> _pages = [
-    ActiveChallengeWidget(),
-    CompletedChallengeWidget()
-  ];
+
+  RestDatasource _api = RestDatasource();
+  List<UserChallenge> userChallenges;
 
   @override
   Widget build(BuildContext context) {
+
+    _api.getUserChallenges(widget.user).then((challenges) {
```

Comment:
For those who may be wondering, this call should really be done on the navigation screen, so that they are loaded by the time you reach here. Also a "loading" status flag should be added so incase they are NOT done, a spinner or some placeholder will be rendered instead